### PR TITLE
Micro optimization multimarker

### LINKED ISSUE #14586: Micro optimizations multimarker
The current codepaths for plotting large scatter dataset using WebGL can be further enhanced.

### PR DESCRIPTION
**Pull request by muendlein**
- [x] issues: fixes #14586
- [x] tests passed

### CODE REVIEW

**Comment by muendlein** [comment]
This PR improves the performance (dragging) for large scatter datasets by 15-25%.

File: bokehjs/src/lib/models/glyphs/webgl/multi_marker.ts [hunk 1]
```diff
@@ -36,7 +36,8 @@ export class MultiMarkerGL extends BaseMarkerGL {
     const nmarkers = main_gl_glyph.nvertices
 
     const ntypes = main_gl_glyph._unique_marker_types.length
-    for (const marker_type of main_gl_glyph._unique_marker_types) {
+    for (let i = 0; i < main_gl_glyph._unique_marker_types.length; i++) {
+      const marker_type = main_gl_glyph._unique_marker_types[i]
```
Comment (mattpap) [inline bokehjs/src/lib/models/glyphs/webgl/multi_marker.ts]: Unless you have data to prove such changes are performance beneficial, I strongly doubt this is a performance optimization. Half a decade ago for sure, but modern browsers can optimize `for of` machine code very well.

**Comment by mattpap** [comment]
Given the extent of changes to FlatBush, I would bring in its units tests, to make sure we aren't breaking some corner cases.

File: bokehjs/src/lib/models/glyphs/webgl/multi_marker.ts (hunk 1 above)
Comment (muendlein) [reply bokehjs/src/lib/models/glyphs/webgl/multi_marker.ts]: @mattpap I have reverted this part now (this seem to have gotten into the commit but was rather part of some play around).

Please note however that the other `for of` optimization brings in some performance uplift for **Chrome** (2x for the paint method, Firefox is not affected). This was also rather surprising to me:

before:

chrome_before_paint

after:

chrome_after_paint

**Comment by muendlein** [comment]
> Given the extent of changes to FlatBush, I would bring in its units tests, to make sure we aren't breaking some corner cases.

@mattpap I have added the two unit tests from FlatBush that are relevant to search.

**Comment by bryevdv** [comment]
The author of Flatbush [just stated an intention today](https://github.com/mourner/flatbush/issues/63#issuecomment-3168500422) to make changes to `searchFn` that would allow us to get rid of `search_apply` entirely in the near future. As such, I am actually 👎 on merging any changes to `search_apply`.

**Comment by bryevdv** [comment]
@muendlein looks there's already a new Flatbush release with the updated `search` function. Do you want to make a PR to update in Bokeh and remove `search_apply` or shall I?

**Comment by muendlein** [comment]
@bryevdv Feel free to go ahead with the FlatBush upgrade.
One thing I like to bring up is that the current implementation of `search_apply` already contains some optimizations which are not included in FlatBush as of right now (the upstream PR is pending for some month already). Therefore some performance regression is expected.

As for this PR I removed everything related to `spatial`, only the loop enhancement in multimarker remains (Chrome specific).

**Review by bryevdv — APPROVED** [review]
